marketdepthviewer/marketdepthviewer.go

```diff
@@ -180,95 +149,30 @@ func handleSubscription(stream api.TradingDataService_ObserveEventBusClient) {
 		for _, event := range eb.Events {
 			switch event.Type {
 			case eventspb.BusEventType_BUS_EVENT_TYPE_MARKET_DATA:
+				mode = event.GetMarketData().MarketTradingMode
 				marketData = event.GetMarketData()
-				drawMarketState()
 			}
 		}
 	}
 }
 
-func initialiseScreen() error {
-	var e error
-	ts, e = tcell.NewScreen()
-	if e != nil {
-		log.Fatalln("Failed to create new tcell screen", e)
-		return e
-	}
-
-	e = ts.Init()
-	if e != nil {
-		log.Fatalln("Failed to initialise the tcell screen", e)
-		return e
-	}
-
-	whiteStyle = tcell.StyleDefault.
-		Background(tcell.ColorReset).
-		Foreground(tcell.ColorWhite)
-	greenStyle = tcell.StyleDefault.
-		Background(tcell.ColorReset).
-		Foreground(tcell.ColorGreen)
-	redStyle = tcell.StyleDefault.
-		Background(tcell.ColorReset).
-		Foreground(tcell.ColorRed)
-
-	return nil
-}
-
-func drawString(x, y int, style tcell.Style, str string) {
-	for i, c := range str {
-		ts.SetContent(x+i, y, c, nil, style)
-	}
-}
-
-func drawHeaders() {
-	w, h := ts.Size()
-
-	// Draw the headings
-	drawString((w/4)-2, 2, whiteStyle, "Bids")
-	drawString((3*w/4)-2, 2, whiteStyle, "Asks")
-
-	drawString((w/4)-19, 3, whiteStyle, "--Volume--")
-	drawString((w/4)+8, 3, whiteStyle, "---Price---")
-	drawString((3*w/4)-22, 3, whiteStyle, "---Price---")
-	drawString((3*w/4)+9, 3, whiteStyle, "--Volume--")
-
-	// If we have a market name, use that
-	if market != nil {
-		text := fmt.Sprintf("Market: %s", market.TradableInstrument.Instrument.Name)
-		drawString(0, 0, whiteStyle, text)
-		drawString(0, 1, whiteStyle, market.Id)
-	} else {
-		text := fmt.Sprintf("Market: %s", market.Id)
-		drawString(0, 0, whiteStyle, text)
+func subscribeMarketDepthSnapshots(dataclient api.TradingDataServiceClient) {
+	req := &api.MarketDepthSubscribeRequest{
+		MarketId: market.Id,
 	}
-	drawString(w-26, 0, whiteStyle, "Last Update Time:")
-
-	drawString((w/4)-8, h-1, whiteStyle, "Volume:")
-	drawString((3*w/4)-8, h-1, whiteStyle, "Volume:")
-}
-
-func drawMarketState() {
-	if marketData != nil {
-		w, h := ts.Size()
-		text := marketData.MarketTradingMode.String()
-		drawString((w-len(text))/2, h-1, whiteStyle, text)
-		text = fmt.Sprintf("Open Interest: %d", marketData.OpenInterest)
-		drawString(w-len(text), 1, whiteStyle, text)
-		ts.Show()
+	stream, err := dataclient.MarketDepthSubscribe(context.Background(), req)
+	if err != nil {
+		log.Fatalln("Failed to subscribe to trades: ", err)
```

Comment:
`Fatal` already calls `os.Exit`. But look at the comment above, about panicking/fataling